scripts/process_jira_tickets.py

```diff
@@ -0,0 +1,247 @@
+    metadata_fields: Dict[str, str],
+    jira_url: str,
+    metadata_field: str,
+):
+    """Prepare documents for embedding using the provided configuration."""
+    documents = []
+
+    for _, row in df.iterrows():
+        processed_data = process_row(
+            row=row,
+            df_columns=df.columns,
+            prefix_fields=prefix_fields,
+            substring_fields=substring_fields,
+            list_fields=list_fields,
+            composite_text_fields=composite_text_fields,
+            metadata_fields=metadata_fields,
+            jira_url=jira_url,
+            metadata_field=metadata_field,
+        )
+        documents.append(processed_data)
+
+    return documents
+
+
+def load_config(config_file):
+    config = ConfigParser()
+    config.read(config_file)
+
+    # Parse configuration sections into variables
+    prefix_fields = dict(config["PrefixFields"]) if "PrefixFields" in config else {}
+    substring_fields = (
+        dict(config["SubstringFields"]) if "SubstringFields" in config else {}
+    )
+
+    # Handle ListFields specially since we need to split the string
+    list_fields = []
+    if "ListFields" in config and "fields" in config["ListFields"]:
+        fields_str = config["ListFields"]["fields"]
+        if fields_str:
+            list_fields = [f.strip() for f in fields_str.split(",")]
+
+    composite_text_fields = (
+        dict(config["CompositeTextFields"]) if "CompositeTextFields" in config else {}
+    )
+    metadata_fields = (
+        dict(config["MetadataFields"]) if "MetadataFields" in config else {}
+    )
+
+    return {
+        "prefix_fields": prefix_fields,
+        "substring_fields": substring_fields,
+        "list_fields": list_fields,
+        "composite_text_fields": composite_text_fields,
+        "metadata_fields": metadata_fields,
+        "jira_url": config["TicketUrl"]["jira_url"],
+        "metadata_field": config["TicketUrl"]["metadata_field"],
+    }
+
+
+def save_to_json_files(data: List[Dict], output_dir: str):
+    """Save each dictionary in the list as a separate JSON file in the specified directory."""
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    for i, item in enumerate(data):
+        output_file = os.path.join(output_dir, f"item_{i}.json")
+        with open(output_file, "w", encoding="utf-8") as f:
+            json.dump(item, f, ensure_ascii=False)
+
+
+@click.command()
+@click.argument("input_file", type=click.Path(exists=True))
+@click.argument("config_file", type=click.Path(exists=True))
+@click.argument("output_dir", type=click.Path())
+def process_data(input_file: str, config_file: str, output_dir: str):
+    """
+    Process data from INPUT_FILE using CONFIG_FILE and save to OUTPUT_FILE.
+
+    The input file should be a CSV file containing the data to process.
+    The config file should be an INI file with the processing configuration.
+    The output file will be saved in JSONL format.
```

Comment:
Clarification: On line #195, looks like the files are being saved as JSON and not JSONL?

Reply:
done